src/AssetManager/View/Helper/Asset.php

```diff
@@ -0,0 +1,115 @@
+<?php
+namespace AssetManager\View\Helper;
+
+use AssetManager\Exception\InvalidArgumentException;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use Zend\View\Helper\AbstractHelper;
+use Zend\Cache\Storage\Adapter\AbstractAdapter as AbstractCacheAdapter;
+
+class Asset extends AbstractHelper
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $serviceLocator;
+
+    /**
+     * @param ServiceLocatorInterface $serviceLocator
+     */
+    public function __construct(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->config         = $serviceLocator->get('config')['asset_manager'];
+    }
+
+    /**
+     * find the file and if it exists, append its unix modification time to the filename
+     *
+     * @param string $assetsPath
+     * @param string $filename
+     * @param string $queryString
+     * @return string
+     */
+    private function elaborateFilePath($assetsPath, $filename, $queryString)
+    {
+        // resolve asset
+        $asset = $this->serviceLocator->get('AssetManager\Service\AssetManager')->getResolver()->resolve($filename);
+        if ($asset !== null) {
+
+            // append last modified date to the filepath and use a custom query string
+            return $filename . '?' . $queryString . '=' . $asset->getLastModified();
+        }
+
+        return $filename;
+    }
+
+    /**
+     * Output the filepath with its unix modification time as query param
+     *
+     * @param string $filename
+     * @return string
+     */
+    public function __invoke($filename)
+    {
+        // search the cache config for the specific file requested (if none, use the default one)
+        if (isset($this->config['caching'][$filename])) {
+            $cacheConfig = $this->config['caching'][$filename];
+        } else if (isset($this->config['caching']['default'])) {
+            $cacheConfig = $this->config['caching']['default'];
+        }
+
+        if (isset($cacheConfig) && isset ($cacheConfig['options']['dir'])) {
```

Comment:
2 remarks.

1. `isset` allows nested values to be supplied. This means that `isset($cacheConfig['options']['dir'])` also works
2. Return early. `if (!isset($cacheConfig['options']['dir'])) { return $filename; }`. Less indentation is happiness